src/dataflow/components/dataflow-program.tsx

```diff
@@ -411,6 +427,39 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     if (plotControl) {
       (plotControl as any).update();
     }
+
+    if (n.name === "Data Storage") {
+      this.updateGraphDataSet(n);
+    }
+  }
+
+  private updateGraphDataSet = (n: Node) => {
+    const recentValuesKey = "recentValues";
+    const recentValues: any = n.data[recentValuesKey];
+    const graphKeys: string[] = [];
+    if (recentValues && recentValues.length) {
+      Object.keys(recentValues[recentValues.length - 1]).forEach((objKey: any) => {
+        graphKeys.push(objKey);
+      });
+    }
+    // make a new dataset
+    const graphDataSet: DataSet = { sequences: [] };
+    if (graphKeys && graphKeys.length) {
+      graphKeys.forEach((graphKey: any) => {
+        // make a new sequence
+        const nameKey = graphKey.replace("num", "sequence");
+        const name: any = n.data[nameKey];
+        const graphSequence: DataSequence = { name, units: "my-units", data: []};
+        recentValues.forEach((recVal: any, i: number) => {
+          const pt: DataPoint = { x: 0, y: 0 };
+          pt.x = i;
+          pt.y = recVal[graphKey] ? recVal[graphKey].val : Number.NaN;
+          graphSequence.data.push(pt);
+        });
+        graphDataSet.sequences.push(graphSequence);
+      });
+    }
+    this.setState ({ graphDataSet });
```

Comment:
`updateGraphDataSet ` is almost certainly going to be removed.  The DF team has requested that we do not show a plot of the 15 second recent values from the Data Storage block and instead that the graph is used to display live and saved datasets.  Nonetheless, we need to put something into the graph until we get to a point where we actually have a dataset that we can hand off to our graph.  All of that is to say I wouldn't worry much about the guts of this function.  It will most likely be removed in a future PR.

Reply:
I added a new piece to the AWS tools in a recent PR, so if you have a `programId` generated from when you hit the Run button, you can now call that function with that original `programId` to fetch all recorded data. Once returned, you can render the data series on the graph. https://github.com/concord-consortium/collaborative-learning/blob/3028cf8eb7f897e6d25f2b1b030c65f4d3a7ffa0/src/dataflow/utilities/aws.ts#L78

The function will return all rows more recent than the timestamp you pass to it, so subsequent calls can be much lighter if you only fetch more recently-recorded data.

Reply:
This is awesome news!  I think I would like to handle it in its own PR, but I'm excited to try to get it working!  Thanks, Chris!